neuron_reduce/subtree_reductor_func.py

```diff
@@ -1,365 +1,437 @@
-        handle_orphan_segments(original_seg_to_reduced_seg, all_segments, vals_per_mech_per_segment, mech_names_per_segment)
-        #print('finish orphan segments fix')
+        handle_orphan_segments(original_seg_to_reduced_seg,
+                               all_segments,
+                               vals_per_mech_per_segment,
+                               mech_names_per_segment)
+
 
-def handle_orphan_segments(original_seg_to_reduced_seg, all_segments, vals_per_mech_per_segment, mech_names_per_segment):
+def handle_orphan_segments(original_seg_to_reduced_seg,
+                           all_segments,
+                           vals_per_mech_per_segment,
+                           mech_names_per_segment):
     ''' This function handle reduced segments that did not had original segments mapped to them'''
-    all_mapped_control_segments = original_seg_to_reduced_seg.values() # Get all reduced segments that have been mapped by a original model segment
+    # Get all reduced segments that have been mapped by a original model segment
+    all_mapped_control_segments = original_seg_to_reduced_seg.values()
     non_mapped_segments = set(all_segments) - set(all_mapped_control_segments)
 
     for reduced_seg in non_mapped_segments:
         seg_secs = list(reduced_seg.sec)
-        #find valid parent
-        parent_seg_index = seg_secs.index(reduced_seg)-1
+        # find valid parent
+        parent_seg_index = seg_secs.index(reduced_seg) - 1
         parent_seg = None
-        while parent_seg_index>-1:
+        while parent_seg_index > -1:
             if seg_secs[parent_seg_index] in all_mapped_control_segments:
                 parent_seg = seg_secs[parent_seg_index]
                 break
             else:
-                parent_seg_index-=1
-        
-        #find valid child
-        child_seg_index = seg_secs.index(reduced_seg)+1
+                parent_seg_index -= 1
+
+        # find valid child
+        child_seg_index = seg_secs.index(reduced_seg) + 1
         child_seg = None
-        while child_seg_index<len(seg_secs):
+        while child_seg_index < len(seg_secs):
             if seg_secs[child_seg_index] in all_mapped_control_segments:
                 child_seg = seg_secs[child_seg_index]
                 break
             else:
-                child_seg_index+=1
+                child_seg_index += 1
 
         if not parent_seg and not child_seg:
             raise Exception("no child seg nor parent seg, with active channels, was found")
-        
+
         if parent_seg and not child_seg:
             for mech in mech_names_per_segment[parent_seg]:
                 reduced_seg.sec.insert(mech)
             for n in vals_per_mech_per_segment[parent_seg]:
-                exec('reduced_seg.' + n +' = np.mean(vals_per_mech_per_segment[parent_seg][n])')
-  
+                setattr(reduced_seg, n, np.mean(vals_per_mech_per_segment[parent_seg][n]))
```

Comment:
removed `exec()` calls

Reply:
I use exec because I got some segfault using setattr in old version of NEURON. 
Well, I assume users should just use the latest

Reply:
interesting, good to know.  I will watch out for that in the future.